// @openrouter/sdk/esm/models/tooldefinitionjson.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 546c340f3013
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { ChatMessageContentItemCacheControl$outboundSchema, } from "./chatmessagecontentitemcachecontrol.js";
export const ToolDefinitionJsonType = {
    Function: "function",
};
/** @internal */
export const ToolDefinitionJsonType$outboundSchema = z.enum(ToolDefinitionJsonType);
/** @internal */
export const ToolDefinitionJsonFunction$outboundSchema = z.object({
    name: z.string(),
    description: z.string().optional(),
    parameters: z.record(z.string(), z.nullable(z.any())).optional(),
    strict: z.nullable(z.boolean()).optional(),
});
export function toolDefinitionJsonFunctionToJSON(toolDefinitionJsonFunction) {
    return JSON.stringify(ToolDefinitionJsonFunction$outboundSchema.parse(toolDefinitionJsonFunction));
}
/** @internal */
export const ToolDefinitionJson$outboundSchema = z.object({
    type: ToolDefinitionJsonType$outboundSchema,
    function: z.lazy(() => ToolDefinitionJsonFunction$outboundSchema),
    cacheControl: ChatMessageContentItemCacheControl$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        cacheControl: "cache_control",
    });
});
export function toolDefinitionJsonToJSON(toolDefinitionJson) {
    return JSON.stringify(ToolDefinitionJson$outboundSchema.parse(toolDefinitionJson));
}
